### DISCUSSION TITLE
Representation of Concept Definition

### DISCUSSION THREAD

**Discussion by AndreyG** [microsoft/ifc-spec#57] [Q&A] [answered]
Let's consider the simplest concepts
```clike
template<typename T> concept A = true;
template<typename T> concept B = A<T>;

template<typename T> using alias = bool_constant<A<T>>;
```
What's the point of representing `B` definition (i.e. `A<T>`) as syntax-tree expression, despite the fact that template argument of `bool_constant` (i.e. the same `A<T>`) is template-id expression?
Do I understand correctly that it's temporary limitation of MSVC and it will be fixed in the future?

**Comment by GabrielDosReis** [comment, marked as answer]
> Do I understand correctly that it's temporary limitation of MSVC and it will be fixed in the future?

Yes, the current representation in MSVC is an unfortunate limitation, and that is definitely NOT the future.  The MSVC team is working on fixing it.  I don't have an ETA at this point, but yes it will be fixed in the future.
